data-model/src/main/java/org/hypertrace/core/datamodel/shared/trace/StructuredTraceBuilder.java

```diff
@@ -420,10 +421,21 @@ public static StructuredTrace buildStructuredTraceFromRawSpans(List<RawSpan> raw
                                                                  String customerId,
                                                                  Timestamps timestamps) {
     Map<String, Entity> entityMap = new HashMap<>();
+    // Relying on insertion ordered keyset
+    LinkedHashMap<Resource, Integer> resourceIndexMap = new LinkedHashMap<>();
```

Comment:
type can be Map interface

Reply:
This was intentional, as we are relying on the properties of a linkedhashmap, not a standard map - they're not interchangeable here (indicated by the comment above).